### ISSUE TITLE
[Tests] Fix RuntimeWarning in lqcontrol.py

### ISSUE THREAD

**Issue by oyamad** [QuantEcon/QuantEcon.py#506]
```
/home/travis/build/QuantEcon/QuantEcon.py/quantecon/lqcontrol.py:244: RuntimeWarning: invalid value encountered in double_scalars
  d = self.beta * np.trace(dot(P, dot(C, C.T))) / (1 - self.beta)
```

**Comment by duncanhobbs** [comment]
I think this is a divide by zero error. `self.beta` (the discount parameter) is set to 1 by default. 

Does it make sense to shift the discount parameter default to something like 0.99?

**Comment by oyamad** [comment]
Thanks @duncanhobbs 

When `T==None` (i.e., the model is one of infinite horizon), then `beta=1` should not be accepted.

It should be checked here: https://github.com/QuantEcon/QuantEcon.py/blob/d9c1e8a1664b422edbeed42cd6065968187de739/quantecon/lqcontrol.py#L137-L146

The pair of default values (`beta=1`, `T=None`) is not a good choice: https://github.com/QuantEcon/QuantEcon.py/blob/d9c1e8a1664b422edbeed42cd6065968187de739/quantecon/lqcontrol.py#L115

@jstac ?

**Comment by jstac** [comment]
I'm a bit confused.  I think `beta=1` is quite common in infinite horizon LQ models, since costs can be driven to zero.

See, e.g.,

https://en.wikipedia.org/wiki/Linear%E2%80%93quadratic_regulator#Infinite-horizon,_discrete-time_LQR

Yet it clearly causes an issue here.  I have to think about this when I have a bit of time.

**Comment by jstac** [comment]
Perhaps we can't have `beta=1` in infinite horizons when there's a stochastic component to the update rule for the state, since it keeps pushing the state away from zero and hence pushing up the cost.

So first best is that we allow `beta=1` when `C=0` and force it to be `< 1` otherwise.  Agree?

**Comment by oyamad** [comment]
@jstac Thanks, but what if `C != 0` and `T != None` (i.e., finite horizon)?

And here (in the method `stationary_values` which is for infinite horizon) https://github.com/QuantEcon/QuantEcon.py/blob/d9c1e8a1664b422edbeed42cd6065968187de739/quantecon/lqcontrol.py#L244 it should be something like:

```py
if self.beta == 1:
    d = 0.
else:
    d = self.beta * np.trace(dot(P, dot(C, C.T))) / (1 - self.beta)
```

**Comment by oyamad** [comment]
Fixed by #509.
